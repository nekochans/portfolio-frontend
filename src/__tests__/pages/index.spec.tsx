/**
 * @jest-environment jsdom
 */
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react';
import IndexPage from '../../pages';

test('IndexPage Snapshot test', () => {
  const testProps = {
    members: [
      {
        id: 1,
        cvUrl: 'https://github.com/keitakn/cv',
        githubPicture:
          'https://avatars1.githubusercontent.com/u/11032365?s=460&v=4',
        githubUserName: 'keitakn',
      },
      {
        id: 2,
        cvUrl: 'https://github.com/kobayashi-m42/cv',
        githubPicture:
          'https://avatars0.githubusercontent.com/u/32682645?s=460&v=4',
        githubUserName: 'kobayashi-m42',
      },
    ],
    webServices: [
      {
        id: 1,
        name: 'Mindexer',
        serviceUrl: 'https://www.mindexer.net',
        ogpUrl: 'https://www.mindexer.net/assets/ogp.png',
        description: 'This service makes Qiita stock convenient.',
      },
      {
        id: 2,
        name: 'LGTMeow',
        serviceUrl: 'https://lgtmeow.com',
        ogpUrl: 'https://lgtmeow.com/ogp.webp',
        description: 'LGTM image share service for cat lovers.',
      },
    ],
  };

  const { asFragment } = render(
    <IndexPage
      members={testProps.members}
      webServices={testProps.webServices}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
