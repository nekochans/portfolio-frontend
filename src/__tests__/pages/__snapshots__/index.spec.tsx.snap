// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage Snapshot test 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header-container w-container"
    >
      <div>
        <a
          href="/"
        >
          <h1>
            nekochans
          </h1>
        </a>
      </div>
    </div>
  </header>
  <section
    class="hero"
  >
    <div
      class="hero-container w-container"
    >
      <h1>
        nekochans
      </h1>
      <p>
        We are a Freelance Web Engineer Organization.
      </p>
      <a
        class="btn"
        href="https://github.com/nekochans"
        rel="noopener noreferrer"
        target="_blank"
      >
        Go To GitHub
      </a>
    </div>
  </section>
  <section
    class="items"
  >
    <div
      class="w-container"
    >
      <h1
        class="heading-decoration"
      >
        Members
      </h1>
      <p>
        Organization members
      </p>
      <div
        class="items-container"
      >
        <article
          class="item"
        >
          <a
            href="https://github.com/keitakn/cv"
            rel="noopener noreferrer"
            target="_blank"
          >
            <figure>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwMCIgaGVpZ2h0PSI3NTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="keitakn"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </figure>
            <h3>
              keitakn
            </h3>
          </a>
        </article>
        <article
          class="item"
        >
          <a
            href="https://github.com/kobayashi-m42/cv"
            rel="noopener noreferrer"
            target="_blank"
          >
            <figure>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwMCIgaGVpZ2h0PSI3NTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="kobayashi-m42"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </figure>
            <h3>
              kobayashi-m42
            </h3>
          </a>
        </article>
      </div>
    </div>
  </section>
  <section
    class="items"
  >
    <div
      class="w-container"
    >
      <h1
        class="heading-decoration"
      >
        WebServices
      </h1>
      <p>
        These are web services developed by the organization
      </p>
      <div
        class="items-container"
      >
        <article
          class="item"
        >
          <a
            href="https://www.mindexer.net"
            rel="noopener noreferrer"
            target="_blank"
          >
            <figure>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwMCIgaGVpZ2h0PSI3NTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Mindexer"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </figure>
            <h3>
              Mindexer
            </h3>
            <p>
              This service makes Qiita stock convenient.
            </p>
          </a>
        </article>
        <article
          class="item"
        >
          <a
            href="https://lgtmeow.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <figure>
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwMCIgaGVpZ2h0PSI3NTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="LGTMeow"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </figure>
            <h3>
              LGTMeow
            </h3>
            <p>
              LGTM image share service for cat lovers.
            </p>
          </a>
        </article>
      </div>
    </div>
  </section>
  <footer
    class="footer"
  >
    <div
      class="footer-container w-container"
    >
      <div
        class="footer-site"
      >
        <h1>
          nekochans
        </h1>
      </div>
      <ul
        class="footer-sns"
      >
        <li>
          <a
            href="https://github.com/nekochans"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="fab fa-github"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
        </li>
      </ul>
      <div
        class="footer-copy"
      >
        Copyright (c) nekochans.
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
